field/composite.go

```diff
@@ -89,6 +90,49 @@ func (f *Composite) SetSpec(spec *Spec) {
 	f.orderedSpecFieldTags = orderedKeys(spec.Subfields, spec.Tag.Sort)
 }
 
+func (f *Composite) GetData(data interface{}) error {
+	rv := reflect.ValueOf(data)
+	if rv.Kind() != reflect.Ptr || rv.IsNil() {
+		return errors.New("data is not a pointer or nil")
+	}
```

Comment:
Should we return nil if `rv.IsNil()`? Will this panic otherwise? Could be worth a unit test

Reply:
@louisheath not sure I understand your concern. We expect that data must be a pointer to the struct, otherwise we can't set values. The trick with `iso8583.Unmarshal` (old `GetData`) is that if field in the struct is not set, it will instantiate an empty value:

```
if dataField.IsNil() {
	dataField.Set(reflect.New(dataField.Type().Elem()))
}
```

Am I missing something here?

Reply:
I was naively drawing parallels from similar implementations which do
```
	if v == nil {
		return nil
	}
```
But it appears the requirements are different here